pkg/apis/storage/types.go

```diff
@@ -672,3 +672,43 @@ type CSIStorageCapacityList struct {
 	// Items is the list of CSIStorageCapacity objects.
 	Items []CSIStorageCapacity
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeAttributesClass represents a class of volume attributes. It holds the mutable
+// attributes of volumes for the provisioner that should create and update volumes.
+//
+// VolumeAttributesClasses are non-namespaced; the name of the volume attributes class
+// according to etcd is in ObjectMeta.Name.
+type VolumeAttributesClass struct {
+	metav1.TypeMeta
+
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta
+
+	// parameters holds the mutable attributes of volumes for the provisioner that should
+	// create and update volumes of this volume attributes class. And these parameters are
+	// immutable.
+	// +optional
+	Parameters map[string]string
```

Comment:
I am reading the wording in CSI spec and here and it appears that:

say I was using a `silver` tier `VolumeAttributeClass` with following parameters:

```
iops: 750000
throughput: 400
```

And then I created a `gold` tier `VolumeAttributeClass` with following parmaeters:

```
throughput: 700
```

Because while I was okay with iops, I needed more throughput. But it seems that now we will call `ControllerModifyVolume` with just `throughput` parameter and that will ideally cause driver to reset iops back to whatever default was (i.e not the values from `silver` VAC but something default). Is that accurate?

Reply:
That's my understanding.

Reply:
The wording is confusing. It might be good to provide an example in the comments to explain what in immutable and what is mutable.

Reply:
@xing-yang added.